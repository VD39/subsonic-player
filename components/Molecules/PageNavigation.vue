<script setup lang="ts">
import ButtonLink from '@/components/Atoms/ButtonLink.vue';
import OverflowShadow from '@/components/Atoms/OverflowShadow.vue';

defineProps<{
  navigation: PageNavigation;
}>();
</script>

<template>
  <nav :class="$style.pageNavigation">
    <OverflowShadow>
      <ul :class="['centerItems', $style.navigation]">
        <template v-for="(path, name) in navigation" :key="name">
          <li :class="$style.item">
            <ButtonLink
              is="nuxt-link"
              class="link pageLink"
              exactActiveClass="pageCurrentLink"
              showText
              :to="path"
            >
              {{ name }}
            </ButtonLink>
          </li>
        </template>
      </ul>
    </OverflowShadow>
  </nav>
</template>

<style module>
.pageNavigation {
  position: relative;
  z-index: 1;
  width: 100%;
}

.navigation {
  --navigation-flex-wrap: nowrap;

  flex: 1;
  flex-wrap: var(--navigation-flex-wrap);
  gap: 0;

  @media (--tablet-up) {
    --navigation-flex-wrap: wrap;
  }

  @media (--mobile-only) {
    border-bottom: 1px solid var(--border-color);
  }
}

.item {
  @media (--mobile-only) {
    flex: 1;
  }
}
</style>
