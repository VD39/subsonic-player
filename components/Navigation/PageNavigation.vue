<script setup lang="ts">
defineProps<{
  navigation: PageNavigation[];
}>();
</script>

<template>
  <nav :class="$style.pageNavigation">
    <ul :class="$style.navigation">
      <li v-for="item in navigation" :key="item.title" data-test-id="item">
        <NuxtLink
          :to="item.to"
          :class="$style.link"
          :exact-active-class="$style.current"
        >
          {{ item.title }}
        </NuxtLink>
      </li>
    </ul>
  </nav>
</template>

<style module>
.pageNavigation {
  margin-bottom: var(--space-24);
}

.navigation {
  display: flex;
  flex-wrap: wrap;
  gap: var(--space-16);
}

.link {
  display: block;
  padding: var(--button-spacing);
}

.current {
  background-color: var(--secondary-background-color);
  border-radius: var(--border-radius-medium);
}
</style>
