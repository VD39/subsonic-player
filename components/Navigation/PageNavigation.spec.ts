import type { VueWrapper } from '@vue/test-utils';
import { mount } from '@vue/test-utils';
import PageNavigation from './PageNavigation.vue';

function factory(props = {}) {
  return mount(PageNavigation, {
    props: {
      navigation: [
        {
          title: 'title',
          to: '/to',
        },
        {
          title: 'title1',
          to: '/to1',
        },
      ],
      ...props,
    },
  });
}

describe('PageNavigation', () => {
  let wrapper: VueWrapper;

  beforeAll(() => {
    wrapper = factory();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows the correct number of links', () => {
    expect(wrapper.findAll('[data-test-id="item"]').length).toBe(2);
  });
});
