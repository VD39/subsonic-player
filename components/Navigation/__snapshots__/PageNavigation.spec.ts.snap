// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageNavigation > matches the snapshot 1`] = `
"<nav class="pageNavigation">
  <ul class="navigation">
    <li data-test-id="item"><a exactactiveclass="current" class="link">title</a></li>
    <li data-test-id="item"><a exactactiveclass="current" class="link">title1</a></li>
  </ul>
</nav>"
`;
